src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Windows.cs

```diff
@@ -14,871 +13,607 @@
 namespace System.Net.Security
 {
     //
-    // The class maintains the state of the authentication process and the security context.
-    // It encapsulates security context and does the real work in authentication and
+    // The class does the real work in authentication and
     // user data encryption with NEGO SSPI package.
     //
     // This is part of the NegotiateStream PAL.
     //
-    internal class NegoState
+    internal partial class NegoState
     {
-        private const int ERROR_TRUST_FAILURE = 1790;   // Used to serialize protectionLevel or impersonationLevel mismatch error to the remote side.
-
-        private static readonly byte[] s_emptyMessage = new byte[0];
-        private static readonly AsyncCallback s_readCallback = new AsyncCallback(ReadCallback);
-        private static readonly AsyncCallback s_writeCallback = new AsyncCallback(WriteCallback);
-
-        private Stream _innerStream;
-        private bool _leaveStreamOpen;
+        internal IIdentity GetIdentity()
+        {
+            CheckThrow(true);
```

Comment:
Please use a named argument for Booleans like this, so it's clear at the call site what the meaning of false and true are, e.g. ```CheckThrow(authSuccessCheck: true);```.